packages/types/src/primitive/Event.ts

```diff
@@ -16,6 +16,11 @@ import Null from './Null';
 import U32 from './U32';
 import Unconstructable from './Unconstructable';
 
+interface ConstructorOptions {
+  DataType?: Constructor<EventData>;
```

Comment:
Same question like last time: do you think it's useful to keep this `DataType` field in the options?

Reply:
I think it makes sense to be able to instanciate an Event/EventRecord and provide it with its DataType, rather than only be able to pass in the whole metadata from which the DataType is extracted

e.g. useful in tests as well

Reply:
I'm still half-convinced about this. For me, from a user's perspective, the user should not handle parts of the metadata, but only the metadata as a whole, and tests should use that too

Not a big deal though, we can still keep it.